### ISSUE TITLE
Extra conversion inserted in an expression tree

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#22261]
This is likely a result of the fix for https://github.com/dotnet/roslyn/issues/4471 which was about the absence of such conversion nodes in general.

Now there is a smaller case where a conversion is unexpected.

```cs
using System;
using System.Linq.Expressions;

namespace TestExpression
{
    class MainClass
    {
        static void Main(string[] args)
        {
            Bar<IFooChild> bar = new Bar<IFooChild>();

            Call(bar);
        }

        static void Call<TInterface>(Bar<TInterface> foo)
            where TInterface : IFoo
        {
            foo.PrintExpression((x) => x.value);
        }
    }

    public interface IFooChild : IFoo
    {
    }

    public interface IFoo
    {
        int value { get; set; }
    }

    public class Bar<TInterface>
        where TInterface : IFoo
    {
        public void PrintExpression(Expression<Func<TInterface, int>> valueExpression)
        {
            Console.WriteLine(valueExpression.ToString());
            return;
        }
    }
} 

```

One workaround is to add 'class' to the template part (change where TInterface : IFoo to where TInterface : class, IFoo)  ,  but ideally compiler should not insert this cast.

**Comment by jcouv** [comment]
@VSadov Does this issue meet the 15.5 bar? If not, move to 15.later. Thanks

**Comment by jcouv** [comment]
@VSadov Does this issue meet the 15.5 bar? If not, move to 15.later. Thanks

**Comment by VSadov** [comment]
I have verified the repro scenario - in both current Roslyn and old naive CSC a cast is inserted. 

I do not think there is a way to fix with without breaking fix for #4471 It looks exactly the case where we were asked to put a cast to match old behavior.

The bug seems to literally asking to undo the change. But then we would break other users....

I think we should continue match native CSC since there is a bigger breaking potential if we deviate from that.
